### PR TITLE
Hw2 task02

### PR DESCRIPTION
**Pull request by mirmozavr**

### CODE REVIEW

File: hw2/task02/task02.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Example 2:
+Input: [2,2,1,1,1,2,2]
+Output: 2, 1
+"""
+from collections import Counter
+
+
+def major_and_minor_elem(numbers: list) -> tuple:
+    counter = Counter(numbers)
+    return counter.most_common()[0][0], counter.most_common()[-1][0]
```
Comment (Cjkjvfnby) [inline hw2/task02/task02.py]: What complexity these operations have?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw2/task02/task02.py (hunk 1 above)
Comment (mirmozavr) [reply hw2/task02/task02.py]: O(N) I assume @Cjkjvfnby

Comment (mirmozavr) [reply hw2/task02/task02.py]: Oh, I see the problem. Are there only 2 unique elements in any given array? I don't see any mention of it. @Cjkjvfnby

Comment (Cjkjvfnby) [reply hw2/task02/task02.py]: > O(N) I assume  

Can you share more details on why do you assume it?

Comment (mirmozavr) [reply hw2/task02/task02.py:20]: the idea behind Counter is same as:
cache = {}
for item in numbers:
    if item in cache:
        cache[item] += 1
    else:
        cache[item] = 1

Comment (mirmozavr) [reply hw2/task02/task02.py]: the idea behind Counter is same as:
```
cache = {}
for item in numbers:
    if item in cache:
        cache[item] += 1
    else:
        cache[item] = 1
```

Comment (Cjkjvfnby) [reply hw2/task02/task02.py]: > the idea behind Counter is same as:

I want to emphasize that this comment is set on a specific line.  All lines before are O(N) but what about this line and the overall complexity of the function?

Comment (mirmozavr) [reply hw2/task02/task02.py]: You mean sorting of dict values that could be O(Nlog(N))? I noticed that Counter is coming out already sorted, so adressing .most_common by indexes is also O(N).
If not, this function could be rewritten using usual dict and manual search for min and max key-value pair

Comment (Cjkjvfnby) [reply hw2/task02/task02.py]: > I noticed that Counter is coming out already sorted,

Can you be more specific, how exactly you have noticed that?  Which method do you use for it?

Comment (mirmozavr) [reply hw2/task02/task02.py]: In experimental manner, by printing it...

Comment (Cjkjvfnby) [reply hw2/task02/task02.py]: At this point, you could just watch into the Counter implementation.

PS. There are 2 methods that responsible for printing the object. __str__ and __repr__.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**